From 4b06724843d4be0b8642d1b0bc299fb90c9ebb9f Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Mon, 14 Jan 2019 22:48:48 +0200
Subject: [PATCH 5/6] common: Add vendor.lineage.touch rules

Change-Id: I2ea88ac5738f09a3cc40ddc8528dde55df8b87dd

# Conflicts:
#	common/vendor/hwservice.te
#	common/vendor/hwservice_contexts
#	common/vendor/system_server.te
---
 common/vendor/hal_lineage_touch_default.te | 11 +++++++++++
 common/vendor/hwservice.te                 |  2 ++
 common/vendor/hwservice_contexts           |  4 ++++
 common/vendor/system_app.te                |  3 +++
 common/vendor/system_server.te             |  1 +
 5 files changed, 21 insertions(+)
 create mode 100644 common/vendor/hal_lineage_touch_default.te
 create mode 100644 common/vendor/system_app.te

diff --git a/common/vendor/hal_lineage_touch_default.te b/common/vendor/hal_lineage_touch_default.te
new file mode 100644
index 0000000..d2f54ae
--- /dev/null
+++ b/common/vendor/hal_lineage_touch_default.te
@@ -0,0 +1,11 @@
+type hal_lineage_touch_default, domain;
+hwbinder_use(hal_lineage_touch_default)
+
+type hal_lineage_touch_default_exec, exec_type, vendor_file_type, file_type;
+init_daemon_domain(hal_lineage_touch_default)
+
+# Check if hwservicemanager is ready
+get_prop(hal_lineage_touch_default, hwservicemanager_prop)
+
+# Add vendor.lineage.touch::* services to hwservicemanager
+add_hwservice(hal_lineage_touch_default, hal_lineage_touch_hwservice)
diff --git a/common/vendor/hwservice.te b/common/vendor/hwservice.te
index 037c0eb..fdb7eb1 100644
--- a/common/vendor/hwservice.te
+++ b/common/vendor/hwservice.te
@@ -2,4 +2,6 @@ type hal_lineage_livedisplay_hwservice, hwservice_manager_type;
 
 type hal_lineage_power_hwservice, hwservice_manager_type;
 
+type hal_lineage_touch_hwservice, hwservice_manager_type;
+
 type hal_lineage_trust_hwservice, hwservice_manager_type;
diff --git a/common/vendor/hwservice_contexts b/common/vendor/hwservice_contexts
index 3ddb8a6..dcb21ab 100644
--- a/common/vendor/hwservice_contexts
+++ b/common/vendor/hwservice_contexts
@@ -3,3 +3,7 @@ vendor.lineage.livedisplay::IColor       u:object_r:hal_lineage_livedisplay_hwse
 vendor.lineage.power::ILineagePower      u:object_r:hal_lineage_power_hwservice:s0
 
 vendor.lineage.trust::IUsbRestrict       u:object_r:hal_lineage_trust_hwservice:s0
+vendor.lineage.touch::IGloveMode             u:object_r:hal_lineage_touch_hwservice:s0
+vendor.lineage.touch::IKeyDisabler           u:object_r:hal_lineage_touch_hwservice:s0
+vendor.lineage.touch::IStylusMode            u:object_r:hal_lineage_touch_hwservice:s0
+vendor.lineage.touch::ITouchscreenGesture    u:object_r:hal_lineage_touch_hwservice:s0
diff --git a/common/vendor/system_app.te b/common/vendor/system_app.te
new file mode 100644
index 0000000..5eabb10
--- /dev/null
+++ b/common/vendor/system_app.te
@@ -0,0 +1,3 @@
+binder_call(system_app, hal_lineage_touch_default)
+
+allow system_app hal_lineage_touch_hwservice:hwservice_manager find;
diff --git a/common/vendor/system_server.te b/common/vendor/system_server.te
index 1b28563..56edda7 100644
--- a/common/vendor/system_server.te
+++ b/common/vendor/system_server.te
@@ -5,3 +5,4 @@ allow system_server hal_lineage_power_hwservice:hwservice_manager find;
 
 allow system_server hal_lineage_trust_hwservice:hwservice_manager find;
 binder_call(system_server, hal_trust_default)
+binder_call(system_server, hal_lineage_touch_default)
-- 
2.17.1

